### PR TITLE
changed URL loading mechanism to get status codes and return errors

### PR DESCRIPTION
**Pull request by dpettigrew**
Relates to https://github.com/walmartlabs-wmusiphone/walmart-ios/pull/1926

### CODE REVIEW

File: Source/Web View/WebViewController.swift [hunk 1]
```diff
@@ -283,7 +283,34 @@ extension WebViewController {
         firstLoadCycleCompleted = false
 
         self.url = url
-        webView.loadRequest(requestWithURL(url))
+        let request = requestWithURL(url)
+
+        NSURLConnection.sendAsynchronousRequest(request, queue: NSOperationQueue.mainQueue()) { (response, data, error) -> Void in
```
Comment (angelodipaolo) [inline Source/Web View/WebViewController.swift]: updating this to use `NSURLSession` should be simple:

```
NSURLSession.sharedSession().dataTaskWithRequest(request) { data, response, error in

}
```

File: Source/Web View/WebViewController.swift [hunk 2]
```diff
@@ -596,15 +623,16 @@ extension WebViewController {
     public func removeErrorDisplay() {
         errorView?.removeFromSuperview()
         errorView = nil
+        showWebView()
     }
    
     /// Override to customize the feature name that appears in the error display.
-    public func featureNameForError(error: NSError) -> String {
+    public func featureNameForError(error: NSError?) -> String {
```
Comment (angelodipaolo) [inline Source/Web View/WebViewController.swift]: hmm why the change to an optional type? we should only be calling this if we actually have a non-nil `NSError` value

File: Source/Web View/WebViewController.swift [hunk 3]
```diff
@@ -573,7 +600,7 @@ extension WebViewController {
 extension WebViewController {
     
     /// Override to completely customize error display. Must also override `removeErrorDisplay`
-     public func renderErrorDisplayWithError(error: NSError, message: String) {
+     public func renderErrorDisplayWithError(error: NSError?, message: String) {
```
Comment (angelodipaolo) [inline Source/Web View/WebViewController.swift]: `error` should be non-optional here as well

File: Source/Web View/WebViewController.swift [hunk 4]
```diff
@@ -142,7 +142,20 @@ public class WebViewController: UIViewController {
     
     /// Set `false` to disable error message UI.
     public var showErrorDisplay = true
-    
+
+    public var userAgent: String?
+
+    public lazy var urlSession: NSURLSession = {
```
Comment (angelodipaolo) [inline Source/Web View/WebViewController.swift]: do you think the session needs to be public? I think we should keep the web view's request loading system internal since the details don't matter to external users and I don't ever see us wanting to allow this to be customized. I think it is better to make this completely private or read-only public

Comment (dpettigrew) [inline Source/Web View/WebViewController.swift]: Probably not. I will change to internal.

Comment (dpettigrew) [inline Source/Web View/WebViewController.swift]: Done
